plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3435,6 +3561,12 @@ private Session prepareCleanUpSession()
         return listAllTableFilesInDirectory(tableDataDir);
     }
 
+    private List<String> getAllMetadataFilesFromTableDirectory(String tableDataDir)
+            throws IOException
+    {
+        return listAllTableFilesInDirectory(Path.of(URI.create(tableDataDir).getPath()));
```

Comment:
`URI.getPath` is bad conversion (lossy), is it `file://` URI?

`Path.of(tableDataDir)` should suffice

Reply:
This is for local file system. Path is from java.nio. It doesn't work with protocol  I've just checked